lib/fs/MiqFsUtil.rb

```diff
@@ -251,28 +232,10 @@ def doUpdate(cs)
           next if cs.exclude && cs.exclude.detect { |e| path =~ e }
           toFile = File.join(destDir, path)
           makePath(File.dirname(toFile))
-          #
-          # If the file path matches an encrypt RE and doesn't
-          # match a noencrypt RE, then encrypt the contents of
-          # the file before copying it to the collection.
-          #
-          if cs.encrypt && cs.encrypt.detect { |e| path =~ e }
-            if !cs.noencrypt || !cs.noencrypt.detect { |ne| path =~ ne }
-              compressFile(path, toFile)
-              next
-            end
-          end
-          #
-          # If the file path matches an compress RE and doesn't
-          # match a nocompress RE, then compress the contents of
-          # the file before copying it to the collection.
-          #
-          if cs.compress && cs.compress.detect { |e| path =~ e }
-            if !cs.nocompress || !cs.nocompress.detect { |ne| path =~ ne }
-              compressFile(path, toFile)
-              next
-            end
-          end
+
+          check_encrypt(cs, i, toFile)
+          check_compress(cs, i, toFile)
+
           copyFile(path, toFile)
```

Comment:
`copyFile` is now called unconditionally. Before this change, it was bypassed via the `next` calls above. Now that the above code has been moved into methods, the `next` calls will no longer work - they are probably not valid in the new methods, as there is no longer a corresponding loop.

Reply:
Yup.  Not really sure what I was thinking.  Let me revisit this.